### ISSUE TITLE
The problem with Cross Modal Attention Mechanism

### ISSUE THREAD

**Issue by MM666-code** [zhao-chunyu/SaliencyMamba#12]
def forward(self, img, feature):
        # 1: get clip feature
        b, c, h, w = feature.shape

        torch_resize = Resize([224, 224])
        img = torch_resize(img)

        with torch.no_grad():
            clip_feature = self.model.encode_image(img)
            # projector: [B, N] ---> [B, C, H/2, W/2]
            clip_feature = clip_feature.view(b, c, h//2, w//2)

        # print(clip_feature.shape)
        # 2: perception matrix
        q = clip_feature.view(b, c, -1)
        k = clip_feature.view(b, c, -1).permute(0, 2, 1)
        # print('q', q.shape, 'k', k.shape)
        perception = torch.bmm(q, k)
        perception = torch.max(perception, -1, keepdim=True)[0].expand_as(perception) - perception
        perception = perception            

Formula 4 in the paper uses a softmax operation to calculate similarity, while in the code, 

"perception=torch. bm (q, k)
perception = torch.max(perception, -1, keepdim=True)[0].expand_as(perception) - perception” 

What are the reasons for the inconsistency between the code and the formula in the paper. Looking forward to receiving your reply.

**Comment by ksk2023** [comment]
Q and K are text information(semantic information).V is the image information, You can see the answer in this web address:
 
https://github.com/zhao-chunyu/SaliencyMamba/blob/main/fig/issue.md
